src/jit/codegenarm.cpp

```diff
@@ -1455,7 +1533,39 @@ void CodeGen::genIntToIntCast(GenTreePtr treeNode)
 //
 void CodeGen::genFloatToFloatCast(GenTreePtr treeNode)
 {
-    NYI("Cast");
+    // float <--> double conversions are always non-overflow ones
+    assert(treeNode->OperGet() == GT_CAST);
+    assert(!treeNode->gtOverflow());
+
+    regNumber targetReg = treeNode->gtRegNum;
+    assert(genIsValidFloatReg(targetReg));
+
+    GenTreePtr op1 = treeNode->gtOp.gtOp1;
+    assert(!op1->isContained());               // Cannot be contained
+    assert(genIsValidFloatReg(op1->gtRegNum)); // Must be a valid float reg.
+
+    var_types dstType = treeNode->CastToType();
+    var_types srcType = op1->TypeGet();
+    assert(varTypeIsFloating(srcType) && varTypeIsFloating(dstType));
+
+    genConsumeOperands(treeNode->AsOp());
+
+    // treeNode must be a reg
+    assert(!treeNode->isContained());
+
```

Comment:
Asserting that the current node (`treeNode`) is not contained isn't really useful, since we would not be generating code for a contained node (it would have done an early-out in `genCodeForTreeNode`). Again, this could be removed in a future change.